MatrixSDK/MXSession.m

```diff
@@ -926,18 +927,35 @@ - (NSString *)syncFilterId
 
 - (BOOL)isPauseable
 {
-    return _state == MXSessionStateSyncInProgress
-        || _state == MXSessionStateRunning
-        || _state == MXSessionStateBackgroundSyncInProgress
-        || _state == MXSessionStatePauseRequested;
+    switch (_state)
```

Comment:
Out of curiosity (beyond the scope of this PR), what are the conventions for using ivars directly as opposed to synthesised getters? (`_state` vs `self.state`)?

Reply:
Tbh we don't have much conventions about them, for the cases we implemented synthesized getter or setters we're trying to ensure they're used instead of ivars. Otherwise there are mixed usage of them.